cpu/nrf52/periph/pwm.c

```diff
@@ -90,9 +90,6 @@ uint32_t pwm_init(pwm_t pwm, pwm_mode_t mode, uint32_t freq, uint16_t res)
 
     /* pin configuration */
     for (unsigned i = 0; i < PWM_CHANNELS; i++) {
-        if (pwm_config[pwm].pin[i] != GPIO_UNDEF) {
-            NRF_P0->PIN_CNF[pwm_config[pwm].pin[i]] = PIN_CNF_SET;
-        }
         /* either left aligned pol or inverted duty cycle */
```

Comment:
```suggestion
        if (pwm_config[pwm].pin[i] == GPIO_UNDEF) {
            continue;
        }
        /* either left aligned pol or inverted duty cycle */
```

should we also do this?

Reply:
I wouldn't, because a) we might be initializing the PWM device from a state that was not its reset state (if a board needs different PWM pins driven at different times, it might just reinit with a different setting), and b) it just adds needless overhead to something that's otherwise just four word writes (at least after I'm done with #15121)

Reply:
Ok, so we always have to init all PWM channels, even if we don't use them, but as long as no output pin is configured, this should have no effect.

Reply:
Yes. And we have to initialize it to be sure that no output pin is set in the first place. (Especially given the Nordic designers' choice that just selecting a pin in a device route already makes it driven, with no need to set it as an output pin in its GPIO properties).

Reply:
And "no effect" is only true as long as the PWM's interrupts stay off, but currently they do, so...